### ISSUE TITLE
How to render to "Camera Far Plane"?

### ISSUE THREAD

**Issue by mrtenda** [RenderHeads/UnityPlugin-AVProVideo#1634]
**Question**

Hi, we're interested in purchasing AVPro Video, but first we need to know if it supports a certain feature we require.

With the default Unity VideoPlayer component, it has a "Render Mode" setting that let me render the video to the ["Camera Far Plane"](https://docs.unity3d.com/560/Documentation/ScriptReference/Video.VideoRenderMode.CameraFarPlane.html), which renders the video "behind the camera's scene". In effect, this basically renders the video behind the scene (instead of the skybox).

Is this something that's possible to do with AVPro Video, and if so, how? I couldn't figure out how to to do this in the free trial version. Thanks.

**Comment by Ste-RH** [comment]
Yes. That is very achievable. The method to implement this will depend on what you are trying to achieve in your use-case. I suggest looking at the ApplyToMaterial and ApplyToMesh components.

**Comment by mrtenda** [comment]
Thanks for the quick reply.

Our use-case is that we want the video to be rendered full-screen behind the scene instead of the skybox.

The video should render to the camera far plane, and should be unaffected by camera movement. Imagine basically showing some 3d Unity characters in front of a youtube video, as if on a green-screen.

What would be the best method to do this? I've looked at the ApplyToMaterial and ApplyToMesh components, but wasn't sure how they'd achieve this. I tried using ApplyToMaterial with the included skybox AVPro Video materials, but those seem to render directly to the skybox, which doesn't really do what we're trying to do here.
